src/Resolution.ts

```diff
@@ -1213,12 +1247,27 @@ export default class Resolution {
       });
     }
 
-    return isApi(config.sourceConfig?.uns)
-      ? new UdApi(config.sourceConfig?.uns)
-      : new Uns(config.sourceConfig?.uns);
+    if (isApi(config.sourceConfig?.uns)) {
+      return new UdApi(config.sourceConfig?.uns);
+    }
+
+    const layer1 =
+      config.sourceConfig?.uns?.locations.Layer1 ||
+      config.sourceConfig?.uns?.locations.eth;
+    const layer2 =
+      config.sourceConfig?.uns?.locations.Layer2 ||
+      config.sourceConfig?.uns?.locations.pol;
+    return layer1 || layer2
```

Comment:
Why is this an OR? If either `layer1` or `layer2` is undefined, wouldn't it be a misconfig?

Reply:
The way our `Uns` is currently configured, it's supposed to work if you define a custom config (Layer1 + Layer2) or if you don't define any config and it's using some internal defaults. 
Internally `Uns` checks if both Layer1 and Layer2 are defined and throws an appropriate error if it's missing. So to avoid re-implementing this logic here (`ensureValidSourceConfig` in `Uns.ts`) I'm checking if anything is defined and passing it to `Uns`.